testsuites/store/src/main/java/org/eclipse/rdf4j/sail/SailConcurrencyTest.java

```diff
@@ -127,7 +127,7 @@ public void run() {
 						conn.addStatement(subject, RDFS.COMMENT, vf.createLiteral("ci" + i), context);
 					}
 					if (rollback) {
-						otherTxnCommitted.await();
+						otherTxnCommitted.await(); // deadlocks here. Waiting for another transaction while holding the exclusive write lock!
 						conn.rollback();
```

Comment:
Hm, I thought I left a reply but it seems to have been swallowed. To repeat: the test is based on the assumption that the sail stack supports concurrent writes - which this reasoner doesn't do. Like @ansell said, the best thing is to ignore/disable this test for this reasoner, and we should clearly document the fact that sail stacks using this reasoner will not fully support concurrent writes.